### DISCUSSION TITLE
How to make relation to self?

### DISCUSSION THREAD

**Discussion by mikield** [laravel/framework#39957] [Q&A]
Hello, I have tried to create relationships based on my business rules to self.

```
public function matchedIncident()
    {
        return $this->hasOne(self::class, "vehicle_plate_license", "vehicle_plate_license")
            ->where('submitted_at', $this->submitted_at)
            ->where('creator_id', '!=', $this->creator_id);
    }
```

Now when I am accessing  `matchedIncident` attrbitue on a `Incident` model instance - I am getting the right result, and everything works.

But when I try to `load` or to use `has` on that relationship - Laravel says that's there is null.

**Comment by erikn69** [comment]
Because `$this->submitted_at` and `$this->creator_id` are null on eager loading, try with [compoships](https://github.com/topclaudy/compoships)
